### PR TITLE
changes for testing statesync on subnet-evm

### PR DESCRIPTION
**Pull request by darioush**
Encountered the following using ANR to test statesync on subnet-evm:
- Allow eth client to connect to different chain IDs
- Allow restart command to specify chain configs
- bump avalanchego version and fix logging types.

### CODE REVIEW

File: cmd/control/control.go [hunk 1]
```diff
@@ -552,6 +552,12 @@ func newRestartNodeCommand() *cobra.Command {
 		"",
 		"whitelisted subnets (comma-separated)",
 	)
+	cmd.PersistentFlags().StringVar(
```
Comment (holisticode) [inline cmd/control/control.go:555]: Wasn't this already done?

File: examples/local/fivenodenetwork/main.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/ava-labs/avalanche-network-runner/local"
 	"github.com/ava-labs/avalanche-network-runner/network"
 	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
```
Comment (holisticode) [inline examples/local/fivenodenetwork/main.go:15]: Note: in this PR https://github.com/ava-labs/avalanche-network-runner/pull/230 we are unifying ANR logging to use just the avalanchego logger (which also used zap, but behind), so depending on what gets merged first, someone might need to address.

File: examples/local/indepth/main.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/ava-labs/avalanchego/config"
 	"github.com/ava-labs/avalanchego/staking"
 	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
```
Comment (holisticode) [inline examples/local/indepth/main.go:17]: Same here

File: local/helpers.go [hunk 4]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/ava-labs/avalanche-network-runner/network/node"
 	"github.com/ava-labs/avalanchego/config"
 	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
```
Comment (holisticode) [inline local/helpers.go:11]: And here

File: rpcpb/rpc.proto [hunk 5]
```diff
@@ -300,6 +300,8 @@ message RestartNodeRequest {
   // Optional fields are set to the previous values if empty.
   optional string exec_path           = 2;
   optional string whitelisted_subnets = 3;
+
+  map<string, string> chain_configs = 4;
```
Comment (holisticode) [inline rpcpb/rpc.proto:304]: I am a bit confused, hadn't this already been done @felipemadero

**Review by holisticode — COMMENTED** [review]
basically lgtm!

But:
* PR https://github.com/ava-labs/avalanche-network-runner/pull/230 is unifying ANR logging to just use the avalanchego logger, and this PR is introducing the zap logger in many places where it was not. Most probably these changes will have to be reverted.
* I am a bit unclear about the changes with chainConfigs, I thought they had been done already (remember to have reviewed something already). Maybe @felipemadero knows better.

File: rpcpb/rpc.proto (hunk 5 above)
Comment (darioush) [reply rpcpb/rpc.proto:304]: this is supported on node start but not sure how it works if you want to restart a node with a different config.

Comment (felipemadero) [reply rpcpb/rpc.proto:304]: Ohh the param is absent from restart node, yes, this should be here. Will also do a review of all of this by tomorrow.

File: cmd/control/control.go (hunk 1 above)
Comment (felipemadero) [reply cmd/control/control.go:555]: it was not

File: examples/local/fivenodenetwork/main.go (hunk 2 above)
Comment (felipemadero) [reply examples/local/fivenodenetwork/main.go:15]: probably we may accept this, 230, and later make a new PR for introducing zap types

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**